apps/heft/src/cli/HeftToolsCommandLineParser.ts

```diff
@@ -222,13 +222,14 @@ export class HeftToolsCommandLineParser extends CommandLineParser {
     process.exitCode = 0;
   }
 
-  private _normalizeCwd(): void {
+  private async _normalizeCwd(): Promise<void> {
     const buildFolder: string = this._heftConfiguration.buildFolder;
     this._terminal.writeLine(`Project build folder is "${buildFolder}"`);
     const currentCwd: string = process.cwd();
     if (currentCwd !== buildFolder) {
       // Update the CWD to the project's build root. Some tools, like Jest, use process.cwd()
       this._terminal.writeVerboseLine(`CWD is "${currentCwd}". Normalizing to project build folder.`);
+      await FileSystem.ensureFolderAsync(this._heftConfiguration.projectHeftDataFolder);
```

Comment:
You're missing the `process.chdir(this._heftConfiguration.projectHeftDataFolder);` part.

Also we should make sure this folder doesn't immediately get deleted again if they're running `heft clean` or `heft build --clean`.